dags/process_zip_dag.py

```diff
@@ -228,6 +230,38 @@ def add_missing_uri_schemes(**context) -> bytes:
     return etree.tostring(article_xml, xml_declaration=True, encoding='UTF-8')
 
 
+def send_article_info_to_details_service(**context) -> bytes:
+    article_xml = get_article_from_previous_task(context)
+    article_id = jats.get_article_id(article_xml)
+
+    categories_in_xml = jats.get_categories(article_xml)
+    category_ids = []
+    for category in categories_in_xml:
+        response = requests.get(
+            '%s/catgories?name=%s' % (DETAILS_SERVICE_URL, quote_plus(category))
+        )
+        response.raise_for_status()
+
+        results = response.json().get('items')
+        if results:
+            category_ids.append(results[0]['id'])
+        else:
+            response = requests.post(
+                '%s/categories' % DETAILS_SERVICE_URL,
+                json={'name': category}
+            )
+            response.raise_for_status()
+            category_ids.append(Path(response.headers['Location']).stem)
```

Comment:
should not assume a particular path format, following the `Location` header leads to a JSON body with a known schema; that is the place to get the ID from
(this makes the API chatty, but it's a trade-off for extensibility as paths and the service on the other side can be changed as long as they conform to the schema)

Reply:
What would be the value of the `Location` header in this instance in your view?

Reply:
Empirically an absolute path but not necessarily containing an hostname:
```
/categories/42
```
or
```
/categories/42/chemistry
```
in a possible alternate implementation.

Reply:
Currently it is in the form of:
```
/categories/42
```
which `pathlib` deals with nicely. The `.stem` property returns all the text after the last `/`.